internal/scheduleringester/schedulerdb.go

```diff
@@ -230,22 +262,27 @@ func (s *SchedulerDb) WriteDbOp(ctx *armadacontext.Context, tx pgx.Tx, op DbOper
 				runAttempted = append(runAttempted, k)
 			}
 		}
-		err := queries.MarkJobRunsFailedById(ctx, runIds)
-		if err != nil {
+		sqlStmt := multiColumnRunsUpdateStmt(updateVals, "failed", "terminated_timestamp", "run_id")
+		if _, err := tx.Exec(ctx, sqlStmt, failTimes...); err != nil {
 			return errors.WithStack(err)
 		}
-		err = queries.MarkJobRunsReturnedById(ctx, returned)
-		if err != nil {
+
+		if err := queries.MarkJobRunsReturnedById(ctx, returned); err != nil {
 			return errors.WithStack(err)
 		}
-		err = queries.MarkJobRunsAttemptedById(ctx, runAttempted)
-		if err != nil {
+		if err := queries.MarkJobRunsAttemptedById(ctx, runAttempted); err != nil {
 			return errors.WithStack(err)
 		}
 	case MarkRunsRunning:
 		runIds := maps.Keys(o)
-		err := queries.MarkJobRunsRunningById(ctx, runIds)
-		if err != nil {
+		updateVals := make([]string, 0, len(runIds))
+		runningTimes := make([]interface{}, 0, len(runIds))
+		for i, runId := range runIds {
+			updateVals = append(updateVals, fmt.Sprintf("('%v'::uuid, true, $%d::timestamptz)", runId, i+1))
```

Comment:
Break this out into separate functions. Where possible, strings should be constants defined at the top of the file instead of created dynamically.